src/vivarium_public_health/population/base_population.py

```diff
@@ -257,22 +257,22 @@ def _format_data_inputs(
         if "year_start" not in scaling_factor.index.names:
             return population_structure, scaling_factor
 
-        # Subset to start year of simulation or closest year
```

Comment:
None of the logic here changed except for calling the new _find_bin_start_index method instead of directly calling np.digitize. Other than that, I just

(1) updated the comment
(2) shuffled the lines around a bit to (I think) make it more clear.